source/extensions/quic_listeners/quiche/envoy_quic_utils.cc

```diff
@@ -118,6 +119,13 @@ Network::ConnectionSocketPtr
 createConnectionSocket(Network::Address::InstanceConstSharedPtr& peer_addr,
                        Network::Address::InstanceConstSharedPtr& local_addr,
                        const Network::ConnectionSocket::OptionsSharedPtr& options) {
+  if (local_addr == nullptr) {
```

Comment:
Should this be loopback?
I'd think
local_addr = Network::Utility::getLocalAddress(host_address->ip()->version());
would make more sense though you may have to do some v6 workarounds for tests as I did in #15027